examples/manipulation/behavior_cloning.py

```diff
@@ -0,0 +1,414 @@
+                action_diff = torch.norm(student_action - teacher_action, dim=-1)
+                condition = (action_diff < 1.0).unsqueeze(-1).expand_as(student_action)
+                action = torch.where(condition, student_action, teacher_action)
+
+                next_obs, reward, done, _ = self._env.step(action)
+                self._cur_reward_sum += reward
+
+                obs = next_obs
+                new_ids = (done > 0).nonzero(as_tuple=False)
+                self._rewbuffer.extend(self._cur_reward_sum[new_ids][:, 0].cpu().numpy().tolist())
+                self._cur_reward_sum[new_ids] = 0
+
+    def save(self, path: str) -> None:
+        """Save model checkpoint."""
+        checkpoint = {
+            "model_state_dict": self._policy.state_dict(),
+            "optimizer_state_dict": self._optimizer.state_dict(),
+            "current_iter": self._current_iter,
+            "config": self._cfg,
+        }
+        torch.save(checkpoint, path)
+        print(f"Model saved to {path}")
+
+    def load(self, path: str) -> None:
+        """Load model checkpoint."""
+        checkpoint = torch.load(path, map_location=self._device, weights_only=False)
+        self._policy.load_state_dict(checkpoint["model_state_dict"])
+        self._optimizer.load_state_dict(checkpoint["optimizer_state_dict"])
+        self.current_iter = checkpoint["current_iter"]
+        print(f"Model loaded from {path}")
+
+    def load_finetuned_model(self, path: str) -> None:
+        """Load a fine-tuned model checkpoint."""
+        checkpoint = torch.load(path, map_location=self._device, weights_only=False)
+        self._policy.load_state_dict(checkpoint["model_state_dict"])
+        self._optimizer.load_state_dict(checkpoint["optimizer_state_dict"])
+        self._current_iter = checkpoint["current_iter"]
+        print(f"Fine-tuned model loaded from {path}")
+
+
+class ExperienceBuffer:
+    """A first-in-first-out buffer for experience replay."""
+
+    def __init__(
+        self,
+        num_envs: int,
+        max_size: int,
+        img_shape: tuple[int, int, int],
+        state_dim: int,
+        action_dim: int,
+        device: str = "cpu",
+    ):
+        self._num_envs = num_envs
+        self._max_size = max_size
+        self._img_shape = img_shape
+        self._state_dim = state_dim
+        self._action_dim = action_dim
+        self._device = device
+        self._ptr = 0
+        self._size = 0
+
+        # Buffers for data
+        self._rgb_obs = torch.empty(max_size, num_envs, *img_shape, dtype=torch.float32, device=device)
+        self._robot_pose = torch.empty(max_size, num_envs, state_dim, dtype=torch.float32, device=device)
+        self._object_poses = torch.empty(max_size, num_envs, 7, dtype=torch.float32, device=device)
+        self._actions = torch.empty(max_size, num_envs, action_dim, dtype=torch.float32, device=device)
```

Comment:
Woud do you make sure this is consistent with policy dtype? What if someone would like to experiment with 64 bits precision during training? Does it make sense?